src/main/java/org/example/game2048/movements/Movement.java

```diff
@@ -0,0 +1,172 @@
+    }
+
+    public static List<Point> moveUp(List<Point> allPointsList) {
+        auxiliaryMethodLeftUp();
+        TEMPORARY_LIST = allPointsList;
+        eachRowProcessing(new XGetter(), new XSetter(), COMPARATOR_Y, new YGetter(), new YSetter());
+        return TEMPORARY_LIST;
+    }
+
+    public static List<Point> moveDown(List<Point> allPointsList) {
+        auxiliaryMethodRightDown();
+        TEMPORARY_LIST = allPointsList;
+        eachRowProcessing(new XGetter(), new XSetter(), COMPARATOR_Y, new YGetter(), new YSetter());
+        return TEMPORARY_LIST;
+    }
+
+
+    private static void oneElementProcessing (Point point, CoordinateGetter coordinateGetter, CoordinateSetter coordinateSetter) {
+        if (coordinateGetter.getCoordinate(point) != firstPlace){
+            coordinateSetter.setCoordinate(point, firstPlace);
+        }
+    }
+
+    private static void twoElementsProcessing (List<Point> twoPointsList, int rowIndex, CoordinateSetter rowSetter, CoordinateSetter coordinateSetter) {
+
+        if (twoPointsList.get(0).getValue().equals(twoPointsList.get(1).getValue())) {
+            rowSetter.setCoordinate(twoPointsList.get(0), rowIndex);
+            coordinateSetter.setCoordinate(twoPointsList.get(0), firstPlace);
+            twoPointsList.get(0).setValue(twoPointsList.get(0).getValue() + twoPointsList.get(1).getValue());
+            TEMPORARY_LIST.remove(twoPointsList.get(1));
+        } else {
+            coordinateSetter.setCoordinate(twoPointsList.get(0), forTwoElementsRowWithoutChangesFirstPlace); //0  2
+            coordinateSetter.setCoordinate(twoPointsList.get(1), forTwoElementsRowWithoutChangesSecondPlace); //1  3
+        }
+    }
+
+    private static void threeElementsProcessing (List<Point> threePointsList, CoordinateSetter rowSetter) {
+
+        threePointsList.forEach(point -> {
+            rowSetter.setCoordinate(point, auxiliaryCoordinateIndex);
+            auxiliaryCoordinateIndex++;
+        });
+
+        if (threePointsList.get(threeElementsFirst).getValue().equals(threePointsList.get(threeElementsSecond).getValue())) {
+            threePointsList.get(threeElementsFirst).setValue(threePointsList.get(threeElementsFirst).getValue() + threePointsList.get(threeElementsSecond).getValue());
+            threePointsList.get(threeElementsSecond).setValue(threePointsList.get(threeElementsThird).getValue());
+            TEMPORARY_LIST.remove(threePointsList.get(threeElementsThird));
+        } else if (threePointsList.get(threeElementsSecond).getValue().equals(threePointsList.get(threeElementsThird).getValue())) {
+            threePointsList.get(threeElementsSecond).setValue(threePointsList.get(threeElementsSecond).getValue() + threePointsList.get(threeElementsThird).getValue());
+            TEMPORARY_LIST.remove(threePointsList.get(threeElementsThird));
+        }
+        auxiliaryCoordinateIndex -= 3;
+    }
+
+    private static void fourElementsProcessing (List<Point> fourElementsList) {
```

Comment:
jest o wiele lepiej i jestem w stanie zaakceptowac takie rozwiazanie ale mozna to napisac lepiej (wiem czepiam sie) :D wyboraz sobie teraz ze przychodzi ktos z biznesu i zwieksza rozmiar planszy z 4 na 5. W Twoim przypadku ktos musial by dopisac tutaj funkcje ktora to przelicza. Jezeli zwiekszymy rozmiar do 20 nie pozostaje nic tylko plakac :D 
a wierze ze mozna to napisac generycznie. Sprobuj opisac slowami co dzieje sie przy ruchu w lewo. Tak na prawde przesuwasz kazdy kwadrat w lewo do momenu az wiecej ruchow nie bedzie mozliwe. I taka definicja pasuje zarowno dla 1 jak i 3 kwadratow. I tak samo moglby wygladac kod :) 

Na szczescie ja nie karze Ci zwiekszac rozmiaru planszy ale mozesz miec to na uwadze i jezeli nastepnym razem w Twoim kodzie zacznie sie pojawiac cos w stylu processOne, processTwo, process... to wiedz ze pewnie mozesz to zastapic jakas petla tylko wystarczy przemyslec rozwiazanie :D

Reply:
Ok got you, but, could you accept exactly this variant in order to we can move forward with project, and of course I'll improve it through the your way because it sounds great.